metabase/metabase
# Milestone 1 Refactor

Epic #35961 

### Description

Refactors #36048 to match the new flow discussed [here](https://metaboat.slack.com/archives/C063Q3F1HPF/p1702298753704459?thread_ts=1701947683.980139&cid=C063Q3F1HPF) and with designs [here](https://www.figma.com/file/6cHI5jaztfjDYvEXEhy9uE/Setup-flow-for-static%2Fpublic-embedding-%26-embedding-feature-discoverability?type=design&node-id=618-4323&mode=design&t=rVe1EGTtAggEf7IR-4).

### How to verify

Go into the Figma designs [here](https://www.figma.com/file/6cHI5jaztfjDYvEXEhy9uE/Setup-flow-for-static%2Fpublic-embedding-%26-embedding-feature-discoverability?type=design&node-id=618-4323&mode=design&t=rVe1EGTtAggEf7IR-4) and see that the flow is replicated on the local environment.

As an admin, you should see:
* A menu every time you click on the share icon. There are no more disabled or other states where the menu wouldn't appear.
* If public sharing is enabled, you should be able to click into the `Create a public link`/`Public link` and see the public link popover.
* If embedding is enabled, you should be able to click on the `Embed` option and see the embed modal
* If one of those settings is disabled, you should see a disabled button with `Embedding/Public links are off`


As a non-admin, you should see:
* If public sharing is enabled and the admin has created a link, you should see the popover
* If public sharing is enabled and the admin has _not_ created a link, you should see a disabled button with "Ask your admin to create a link"
* If public sharing is disabled, you should see a disabled button with "Public links are disabled"


### Checklist
